scripts/utils/metrics.py

```
@@ -0,0 +1,141 @@
+import boto3  # type: ignore
+from datetime import datetime
+import json
+from pathlib import Path
+import re
+from typing import Optional
+
+
+def report_metrics(*,
+                   run_stdout: str,
+                   run_start_time: datetime,
+                   run_end_time: datetime,
+                   s3_client_id: str,
+                   workload_path: Path,
+                   bucket: str,
+                   region: str,
+                   target_throughput_Gbps: float,
+                   instance_type: Optional[str],
+                   branch: Optional[str],
+                   ):
+    # parse stdout
+    throughput_per_run_Gbps = _parse_throughput_per_run_in_gigabits(run_stdout)
+    run_count = len(throughput_per_run_Gbps)
+
+    # bail out if no successful runs
+    if run_count == 0:
+        return
+
+    # get data from workload
+    with open(workload_path) as f:
+        workload = json.load(f)
```

should we fail if workload path is invalid?

---

anything that runs this would have failed earlier, and we already have nice error handling there:
https://github.com/awslabs/aws-crt-s3-benchmarks/blob/2527575fe38ce02c4db1db64c52d793bb2a6d4d4/scripts/utils/__init__.py#L26-L27